tonic/src/extensions.rs

```diff
@@ -0,0 +1,67 @@
+use std::fmt;
+
+/// A type map of protocol extensions.
+///
+/// `Extensions` can be used by [`Interceptor`] and [`Request`] to store extra data derived from
+/// the underlying protocol.
+///
+/// [`Interceptor`]: crate::Interceptor
+/// [`Request`]: crate::Request
+pub struct Extensions(http::Extensions);
```

Comment:
nit (not blocking): Should we make this a proper struct with an `inner`? Idk what style I like better honestly but I tend to do that over a tuple struct 🤷 up to you doesn't matter too much.

Reply:
I don't really have a preference. I'll change it.